### PR TITLE
Replace FilClient with Lassie RetrievalClient

### PR DESCRIPTION
**Pull request by kylehuntsman**
This PR implements the Lassie RetrievalClient as a replacement for FilClient. The goal being to replace the instantiation of FilClient with Lassie's RetrievalClient. In doing so, there were a couple updates required to provide some functionality that FilClient provided that the Lassie RetrievalClient did not.

#### MinerPeerGetter
In the[ `EstuaryEndpoint`](https://github.com/application-research/autoretrieve/blob/feat%2Flassie-update/endpoint/estuary.go#L68) and the [`Config`](https://github.com/application-research/autoretrieve/blob/feat%2Flassie-update/autoretrieve.go#L175-L192), FilClient was being passed around to use it's `api.MinerPeer()` function. A [`MinerPeerGetter`](https://github.com/application-research/autoretrieve/blob/feat/lassie-update/paychannelmanager/lassiepaychannelmanager.go) abstraction was made to pass around _just_ the `MinerPeer()` function instead of all of FilClient.

#### LassiePayChannelManager
One of the goals of Lassie was to remove all dependencies on Lotus and FilClient. In doing so, Lassie's RetrievalClient does not implement a `PayChannelManager` which is required to make paid retrievals. The [`LassiePayChannelManager`](https://github.com/application-research/autoretrieve/blob/feat%2Flassie-update/paychannelmanager/lassiepaychannelmanager.go) implementation was created to provide `PayChannelManager` functionality. The implementation of the [`GetPayChannelWithMinFunds()`](https://github.com/application-research/autoretrieve/blob/feat%2Flassie-update/paychannelmanager/lassiepaychannelmanager.go#L32-L69) method and the [`payChannelApi`](https://github.com/application-research/autoretrieve/blob/feat%2Flassie-update/paychannelapi.go) struct were taken directly from FilClient as they were originally a non exported function and struct.

#### Removal of LogRetrievals options
This was an early addition to Autoretrieve when we were still running it via EC2 instance and watching terminal logs. This was the last thing using the FilClient client for it's LibP2PTransferManager. After discussing with @elijaharita to make sure Estuary wasn't relying on this logging, it was removed to be able to completely remove the FilClient client.

#### Copy of FilClient `MsgPusher` and `Keystore`
The last two FilClient dependencies that Autoretrieve depended on were the `MsgPusher` and `Keystore`. These two structs did not rely on anything else in FilClient. To completely remove the dependency on FilClient, these two structs were copy-pasta'd from FilClient and into Autoretrieve.

### CODE REVIEW

File: autoretrieve.go [hunk 1]
```diff
@@ -134,45 +133,51 @@ func New(cctx *cli.Context, dataDir string, cfg Config) (*Autoretrieve, error) {
...
 
-	walletAddr, err := wallet.GetDefault()
-	if err != nil {
-		logger.Warnf("Could not load any default wallet address, only free retrievals will be attempted: %v", err)
-		walletAddr = address.Undef
-	} else {
-		logger.Infof("Using default wallet address %s", walletAddr)
+	// Set up the PayChannelManager
+	ctx, shutdown := context.WithCancel(context.Background())
+	mpusher := filclient.NewMsgPusher(api, wallet)
```
Comment (rvagg) [inline autoretrieve.go:138]: so we're not actually getting rid of filclient here, we're still using it to handle messages needed by the payment channel manager; is that a better state than outsourcing payment channels to filclient still? seems like either way we're in a half-way position.

the message pusher is just interacting with the api, so maybe we just copy that across too? https://github.com/application-research/filclient/blob/master/msgpusher.go#L25

basically, if we're trying to remove filclient, shouldn't we want to see it disappear from imports and go.mod?

Comment (kylehuntsman) [reply autoretrieve.go]: Latest commit copies the filclient MsgPusher and Keystore into Autoretrieve.

**Review by rvagg — APPROVED** [review]

**Comment by kylehuntsman** [comment]
I can't tell what version of staticcheck GitHub actions are running, but upgrading to staticcheck to 2022.1.3 on my local machine fixed the check.
